wgpu/tests/common/image.rs

```diff
@@ -71,80 +65,172 @@ pub fn calc_difference(lhs: u8, rhs: u8) -> u8 {
     (lhs as i16 - rhs as i16).unsigned_abs() as u8
 }
 
+fn add_alpha(input: &[u8]) -> Vec<u8> {
+    input
+        .chunks_exact(3)
+        .flat_map(|chunk| [chunk[0], chunk[1], chunk[2], 255])
+        .collect()
+}
+
+fn remove_alpha(input: &[u8]) -> Vec<u8> {
+    input
+        .chunks_exact(4)
+        .flat_map(|chunk| &chunk[0..3])
+        .copied()
+        .collect()
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn print_flip(pool: &mut nv_flip::FlipPool) {
+    println!("\tMean: {:.6}", pool.mean());
+    println!("\tMin Value: {:.6}", pool.min_value());
+    for percentile in [25, 50, 75, 95, 99] {
+        println!(
+            "\t      {percentile}%: {:.6}",
+            pool.get_percentile(percentile as f32 / 100.0, true)
+        );
+    }
+    println!("\tMax Value: {:.6}", pool.max_value());
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum ComparisonType {
+    // If the mean error is greater than the given value, the test will fail.
+    Mean(f32),
+    // If the error at the given percentile [0, 1] is greater than the given value, the test will fail.
+    Percentile(f32, f32),
```

Comment:
nitpick: I'd rather the two fields be named (I don't think I'll remember which is which in a couple of days).

Also I'm not sure I parse the comment: does `[0, 1]` refer to the range of the error?